File: workers/worker-jvm/src/main/java/com/openregex/Registry.java
package com.openregex;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.openregex.engines.Engines;
import redis.clients.jedis.Jedis;

public class Registry {
    private static final ObjectMapper mapper = new ObjectMapper();

    public static void registerEngines(Jedis jedis) throws Exception {
        jedis.hset(
                "openregex:workers",
                Engines.WORKER_INFO.worker_name(),
                mapper.writeValueAsString(Engines.WORKER_INFO)
        );
        System.out.printf("[Worker] Registered '%s' with %d engines.%n",
                Engines.WORKER_INFO.worker_name(),
                Engines.WORKER_INFO.engines().size()
        );
    }
}

File: workers/worker-c-cpp/src/worker_c_cpp/registry.py
import redis
from worker_c_cpp.engines import WORKER_INFO

def register_engines(redis_client: redis.Redis):
    """Registers the C/C++ worker and its engines to the Backend via Redis."""
    redis_client.hset(
        "openregex:workers",
        WORKER_INFO.worker_name,
        WORKER_INFO.model_dump_json()
    )
    print(f"[Worker] Registered '{WORKER_INFO.worker_name}' with {len(WORKER_INFO.engines)} engines.", flush=True)